modules/salt/pages/salt-gitfs.adoc

```diff
@@ -0,0 +1,67 @@
+[[salt.gitfs]]
+= The ``gitfs`` Fileserver Backend
+
+In {productname}, [package]``pygit2`` is the supported Python interface to git.
+When [package]``pygit2`` is installed the ``gitfs`` fileserver backend is available and it is a supported feature.
+
+
+Configuration options are set in the [path]``/etc/salt/master`` file or a separate configuration file [path]``/etc/salt/master.d/``.  Basic settings are:
+
+
+fileserver_backend::
+List of fileserver backends that the Salt master checks for files in the order they are defined.  Options:
++
+* [literal]``roots``: Files local on the Salt master ({productname} Server).
+[literal]``roots`` is required to keep the product running.
+You can only enable [literal]``gitfs`` optionally.
+Additionally, {suse} strongly recommends to prefer [literal]``roots`` (local files) over [literal]``gitfs``.
+The standard backend.
+* [literal]``gitfs``: Files stored in one or more git repositories.
+The repositories are defined with [literal]``gitfs_remotes``.
++
+Example:
++
+----
+fileserver_backend:
+  - roots
+  - git
+----
+
+gitfs_remotes::
+List of git repositories.
+``git://``, ``https://``, ``file://``, or ``ssh://`` URLs can be configured.
+For SSH remotes, a [command]``scp``-like syntax is also supported; for example:
+[literal]``gitlab@gitlab.example.com:universe/setup.git``.
+Then you can also specify options for credentials, file locations, or branches such as [literal]``pubkey``, [literal]``privkey``, [literal]``root``,[literal]``base``.
++
+Example:
++
+----
+gitfs_remotes:
+  - https://example.com/myformulas/formula.git
+  - gitlab@gitlab.example.com:universe/setup.git:
+    - pubkey: /var/lib/salt/.ssh/id_rsa_gitlab.pub
+    - privkey: /var/lib/salt/.ssh/id_rsa_gitlab
+    - root: srv/salt
+    - base: master
+----
+
+ext_pillar::
+List of external pillar interfaces.
+Salt can also serve pillar data from one or more git repositories.
+For syntax and options, also see the [literal]``gitfs_remotes`` setting.
++
+Example:
++
+----
+ext_pillar:
+  - git:
+    - master gitlab@gitlab.example.com:universe/setup.git:
+      - root: srv/pillar
+      - pubkey: /var/lib/salt/.ssh/id_rsa_gitlab.pub
+      - privkey: /var/lib/salt/.ssh/id_rsa_gitlab
+----
```

Comment:
Can we come up with a tidier IA here? Maybe a table?

Reply:
I think simple lists with code examples are easier to read.  This is not the (new) reference guide ;)
[side note: writing adoc tables isn't fun.]

Reply:
I hate deflists more than I hate tables, I guess ;) I'll leave this in your capable hands, so do whatever you think works best.

Reply:
ok, thanks!  if it helps we can later make use of the table style.